Polygon_repair_2/include/CGAL/Polygon_repair_2/Polygon_repair_2.h

```diff
@@ -82,6 +82,47 @@ class Polygon_repair_2 {
   typedef CGAL::Constrained_Delaunay_triangulation_2<Kernel, Triangulation_data_structure, Tag> Constrained_Delaunay_triangulation;
   typedef Triangulation_with_odd_even_constraints_2<Constrained_Delaunay_triangulation> Triangulation;
 
+  struct Polygon_compare {
+    bool operator()(const Polygon_2<Kernel, PolygonContainer>& pa, const Polygon_2<Kernel, PolygonContainer>& pb) const {
+      typename Polygon_2<Kernel, PolygonContainer>::Vertex_iterator va = pa.vertices_begin();
+      typename Polygon_2<Kernel, PolygonContainer>::Vertex_iterator vb = pb.vertices_begin();
+      while (va != pa.vertices_end() && vb != pb.vertices_end()) {
+        if (*va != *vb) return *va < *vb;
+        ++va;
+        ++vb;
+      } if (vb == pb.vertices_end()) return false;
+      return true;
+    }
+  };
+
+  struct Polygon_with_holes_compare {
+    bool operator()(const Polygon_with_holes_2<Kernel, PolygonContainer>& pa, const Polygon_with_holes_2<Kernel, PolygonContainer>& pb) const {
+      typename Polygon_2<Kernel, PolygonContainer>::Vertex_iterator va = pa.outer_boundary().vertices_begin();
+      typename Polygon_2<Kernel, PolygonContainer>::Vertex_iterator vb = pb.outer_boundary().vertices_begin();
+      while (va != pa.outer_boundary().vertices_end() && vb != pb.outer_boundary().vertices_end()) {
+        if (*va != *vb) return *va < *vb;
+        ++va;
+        ++vb;
+      } if (vb != pb.outer_boundary().vertices_end()) return true;
+      else if (va != pa.outer_boundary().vertices_end()) return false;
+      typename Polygon_with_holes_2<Kernel, PolygonContainer>::Hole_const_iterator ha = pa.holes_begin();
+      typename Polygon_with_holes_2<Kernel, PolygonContainer>::Hole_const_iterator hb = pb.holes_begin();
+      while (ha != pa.holes_end() && hb != pb.holes_end()) {
```

Comment:
Apply the operator of `Polygon_compare`  instead of duplicating code.

Reply:
Sure, I can do it, but it would be slower since I'd need to do the comparison twice (a < b and b < a) in some cases